deep_field_metadetect/jaxify/jax_metacal.py

```diff
@@ -0,0 +1,636 @@
+
+    if kim_size is None:
+        kim = psf.drawKImage(nx=4 * nxy_psf, ny=4 * nxy_psf, scale=dk)
+    else:
+        kim = psf.drawKImage(nx=kim_size, ny=kim_size, scale=dk)
+
+    # This will lead to a differnce in reconv psf size between GS and JGS
+
+    karr_r = kim.real.array
+    # Find the smallest r where the kval < small_kval
+    nk = karr_r.shape[0]
+    kx, ky = jnp.meshgrid(jnp.arange(-nk / 2, nk / 2), jnp.arange(-nk / 2, nk / 2))
+    ksq = (kx**2 + ky**2) * dk**2
+    ksq_max = jnp.min(jnp.where(karr_r < small_kval * psf.flux, ksq, jnp.inf))
+
+    # We take our target PSF to be the (round) Gaussian that is even smaller at
+    # this ksq
+    # exp(-0.5 * ksq_max * sigma_sq) = smaller_kval
+    sigma_sq = -2.0 * jnp.log(smaller_kval) / ksq_max
+
+    dilation = 1.0 + 2.0 * step
+    return jax_galsim.Gaussian(sigma=jnp.sqrt(sigma_sq) * dilation).withFlux(flux)
+
+
+@partial(jax.jit, static_argnames=["dk", "nxy_psf"])
+def jax_get_gauss_reconv_psf(dfmd_obs, nxy_psf, dk, step=DEFAULT_STEP):
+    """Get the Gaussian reconv PSF for a DFMdetObs."""
+    psf = get_jax_galsim_object_from_dfmd_obs_nopix(dfmd_obs.psf, kind="image")
+    return jax_get_gauss_reconv_psf_galsim(psf, nxy_psf=nxy_psf, dk=dk, step=step)
+
+
+@partial(jax.jit, static_argnames=["nxy_psf", "scale"])
+def jax_get_max_gauss_reconv_psf_galsim(
+    psf_w, psf_d, nxy_psf, scale=0.2, step=DEFAULT_STEP
+):
+    """Get the larger of two Gaussian reconvolution PSFs for two galsim objects."""
+    dk = compute_stepk(pixel_scale=scale, image_size=nxy_psf)
+    mc_psf_w = jax_get_gauss_reconv_psf_galsim(psf_w, dk, nxy_psf, step=step)
+    mc_psf_d = jax_get_gauss_reconv_psf_galsim(psf_d, dk, nxy_psf, step=step)
+
+    return jax.lax.cond(
+        mc_psf_w.fwhm > mc_psf_d.fwhm, lambda: mc_psf_w, lambda: mc_psf_d
+    )
+
+
+@partial(jax.jit, static_argnames=["scale", "nxy_psf"])
+def jax_get_max_gauss_reconv_psf(obs_w, obs_d, nxy_psf, scale=0.2, step=DEFAULT_STEP):
+    """Get the larger of two reconv PSFs for two DFMdetObservations."""
+    psf_w = get_jax_galsim_object_from_dfmd_obs_nopix(obs_w.psf, kind="image")
+    psf_d = get_jax_galsim_object_from_dfmd_obs_nopix(obs_d.psf, kind="image")
+    return jax_get_max_gauss_reconv_psf_galsim(
+        psf_w, psf_d, nxy_psf, scale=scale, step=step
+    )
+
+
+@partial(jax.jit, static_argnames=["nxy_psf", "max_min_fft_size"])
+def _jax_render_psf_and_build_obs(
+    image, dfmd_obs, reconv_psf, nxy_psf, weight_fac=1, max_min_fft_size=1024
+):
+    reconv_psf = reconv_psf.withGSParams(
+        minimum_fft_size=max_min_fft_size,
+        maximum_fft_size=max_min_fft_size,
+    )
+
+    pim = reconv_psf.drawImage(
+        nx=nxy_psf,
+        ny=nxy_psf,
+        wcs=dfmd_obs.psf.wcs._local_wcs,
+        offset=jax_galsim.PositionD(
+            x=dfmd_obs.psf.wcs.origin.x - nxy_psf / 2,
+            y=dfmd_obs.psf.wcs.origin.y - nxy_psf / 2,
+        ),
+    ).array
+
+    obs_psf = dfmd_obs.psf._replace(image=pim)
```

Comment:
```suggestion
    # the _replace calls are standard methods on namedtuples
    obs_psf = dfmd_obs.psf._replace(image=pim)
```

Add this comment for clarity.

Reply:
Skipping this because dfmd_obs.psf is no longer a namedtuple, and replace function is defined in jaxify.observation